source/vulkan/vulkan_impl_command_list.cpp

```diff
@@ -864,43 +865,169 @@ void reshade::vulkan::command_list_impl::resolve_texture_region(api::resource sr
+		dst_subresource_range.layerCount = resolve_region.dstSubresource.layerCount;
+
+		VkImageViewUsageCreateInfo usage_info = { VK_STRUCTURE_TYPE_IMAGE_VIEW_USAGE_CREATE_INFO };
+		usage_info.usage = VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT;
+
+		VkImageViewCreateInfo img_create_info = { VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO, &usage_info };
+		img_create_info.image = dst.handle;
+		img_create_info.viewType = VK_IMAGE_VIEW_TYPE_2D_ARRAY;
+		img_create_info.format = src_data->create_info.format;
+		img_create_info.subresourceRange = dst_subresource_range;
+
+		VkImageView dst_img_view;
+		if (vk.CreateImageView(_device_impl->_orig, &img_create_info, nullptr, &dst_img_view)) {
+			assert(false);
+			return;
+		}
 
-	region.dstOffset = { dst_x, dst_y, dst_z };
+		VkImageSubresourceRange src_subresource_range;
+		src_subresource_range.aspectMask = resolve_region.srcSubresource.aspectMask;
+		src_subresource_range.baseMipLevel = resolve_region.srcSubresource.mipLevel;
+		src_subresource_range.levelCount = 1;
+		src_subresource_range.baseArrayLayer = resolve_region.srcSubresource.baseArrayLayer;
+		src_subresource_range.layerCount = resolve_region.srcSubresource.layerCount;
 
-	vk.CmdResolveImage(_orig,
-		(VkImage)src.handle, VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
-		(VkImage)dst.handle, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,
-		1, &region);
+		img_create_info.image = src.handle;
+		img_create_info.subresourceRange = src_subresource_range;
+
+		VkImageView src_img_view;
+		if (vk.CreateImageView(_device_impl->_orig, &img_create_info, nullptr, &src_img_view)) {
+			assert(false);
+			return;
+		}
+
+		VkRenderingAttachmentInfo depth_attachment = { VK_STRUCTURE_TYPE_RENDERING_ATTACHMENT_INFO };
+		depth_attachment.imageView = src_img_view;
+		depth_attachment.imageLayout = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
+		depth_attachment.resolveMode = VK_RESOLVE_MODE_AVERAGE_BIT;
+		depth_attachment.resolveImageView = dst_img_view;
+		depth_attachment.resolveImageLayout = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
+		depth_attachment.loadOp = VK_ATTACHMENT_LOAD_OP_LOAD;
+		depth_attachment.storeOp = VK_ATTACHMENT_STORE_OP_STORE;
+
+		VkRenderingAttachmentInfo stencil_attachment = depth_attachment;
+		stencil_attachment.resolveMode = VK_RESOLVE_MODE_SAMPLE_ZERO_BIT;
+
+		VkExtent3D extent = src_extent;
+
+		VkRenderingInfo rendering_info = { VK_STRUCTURE_TYPE_RENDERING_INFO };
+		rendering_info.renderArea.offset = VkOffset2D { 0, 0 };
+		rendering_info.renderArea.extent = VkExtent2D { extent.width, extent.height };
+		rendering_info.layerCount = resolve_region.dstSubresource.layerCount;
+
+		if (src_subresource_info.aspectMask & VK_IMAGE_ASPECT_DEPTH_BIT)
+			rendering_info.pDepthAttachment = &depth_attachment;
+
+		if (dst_subresource_info.aspectMask & VK_IMAGE_ASPECT_STENCIL_BIT)
+			rendering_info.pStencilAttachment = &stencil_attachment;
+
+		vk.CmdBeginRendering(_orig, &rendering_info);
+		end_render_pass();
```

Comment:
This should simply be `vk.CmdEndRendering(_orig);` instead of `end_render_pass();` (which does more than is needed here).